[ADD] practical: new doc about database upgrade (#809)
odoo/documentation-user: db_management/db_upgrade.rst

functionality are still correct and working as expected.

If you do find discrepancies, you'll be able to:

* | :ref:`Report your issues <db-upgrade/test-assistance>`
  | and/or
* Ask for a new :ref:`test request <db-upgrade/test-db-request>` after the reported issues have
  been fixed in the upgrade script.

When you do not find any discrepancies, you'll be able to:

* Move on to the upgrade of your :ref:`production database <db-upgrade/production-live>`.

.. _db-upgrade/test-db-request:

Request a test database
~~~~~~~~~~~~~~~~~~~~~~~

When filling the `website form <https://upgrade.odoo.com>`_, select *Testing* purpose.

.. image:: media/db-upgrade-test-purpose.png
   :align: center
   :alt: Selection of the "Testing" purpose in the upgrade form on Odoo

.. _db-upgrade/test-guidance:

Test guidance
~~~~~~~~~~~~~

Every business and organization has its own operational needs and will have to test its specific
Odoo instance respectively. However, we recommend you look at `the test scenario
<https://drive.google.com/open?id=1Lm4JqbsHBirB1wMi14UChoz_YHLjx5ec>`_ we created, a high-level idea
of what you should test and look out for.

> .. todo:: change link "test scenario" once the related doc is published

Reviewer: Shouldn't it be done before merging?

Author: @KangOl 
it's another doc that isn't in the repo yet.
So, until then, the google link is the valid one.

      :alt: Selection of "An issue related to my upgrade" as Ticket Type in the support form on Odoo

   .. warning::
      If you choose another *Ticket Description* type, the request will be redirected to another
      team than the upgrade one and will slow down the processing and response time.

#. Please provide as much detail as you can. Where applicable, illustrate the current and previous
   flows with videos and/or screenshots. This will avoid clarifying questions and speed up the
   resolution process significantly.

   .. image:: media/db-upgrade-test-assistance-details.png
      :align: center
      :alt: "Detailed Description" field in the support form on Odoo

.. note::
   * The purpose of the test phase is not to correct existing data or configurations in your
     database.
   * |assistance-contact|

.. _db-upgrade/steps-production:

The production launch
---------------------

.. _db-upgrade/production-live:

Production goes live
~~~~~~~~~~~~~~~~~~~~

The production upgrade request is when you decide to upgrade your current database with all your
production data (invoices, VAT returns, inventories, current orders) to a new version of your choice.

> After your :ref:`tests <db-upgrade/steps-test>` are completed to your satisfaction, submiy the

Reviewer: Typo
```suggestion
After your :ref:`tests <db-upgrade/steps-test>` are completed to your satisfaction, submit the
```

Author: thanks :)
it's fixed

:banner: banners/db-upgrade.png

.. _db-upgrade:

=======
Upgrade
=======

.. _db-upgrade/overview:

Overview
========

.. _db-upgrade/process:

The upgrade process
-------------------

This upgrade page is for our On-Premise (self-hosted) and Odoo.sh customers. If you are hosted
Online, please check our :ref:`instruction page for our Online (SaaS) customers <upgrade_button>`.

.. _db-upgrade/definition:

Definition
~~~~~~~~~~

An upgrade is switching to a newer version of Odoo (e.g., Odoo 13.0 to Odoo 14.0)

An upgrade does not cover:

> * changing :ref:`Editions <db-upgrade/editions-change>` (i.e., Community to Enterprise edition)

Author: okay! I thought my refs were too long already.
Also, I thought that having the doc title in the ref was enough.
Most of the other refs in the doc don't follow a "path". Some are way too short and don't help much indeed.
I change it and add the "faq" to all that apply

* switching :ref:`hosting type <db-upgrade/hosting-types-switch>` (i.e., On-Premise to Online or
  Odoo.sh)
* migration from another ERP to Odoo

.. note::
   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact

>    our `Sales department <mailto:sales@odoo.com>`_.

Author: @avg-odoo 
you tell me.
Should I delete the note "contact your sales" in this part?

```
Definition
~~~~~~~~~~

An upgrade is switching to a newer version of Odoo (e.g., Odoo 13.0 to Odoo 14.0)

An upgrade does not cover:

* changing :ref:`Editions <db-upgrade/faq/editions-change>` (i.e., Community to Enterprise edition)
* switching :ref:`hosting type <db-upgrade/faq/hosting-types-switch>` (i.e., On-Premise to Online or
  Odoo.sh)
* migration from another ERP to Odoo

.. note:: |assistance-contact|
```

The Upgrade Service is limited to the technical conversion and adaptation of your database (standard
modules and data) to make it compatible with the targeted Version.

What is NOT covered?
~~~~~~~~~~~~~~~~~~~~

* The cleaning of pre-existing data & configuration while upgrading
* Any new developments and/or upgrade of your own :ref:`custom modules <db-upgrade/custom-modules>`
* `Training <https://www.odoo.com/learn>`_ on the new version

You can get more information about your Enterprise Licence on our :ref:`Odoo Enterprise Subscription
Agreement <upgrade>` page.

.. note::
   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
   our `Sales department <mailto:sales@odoo.com>`_.

.. _db-upgrade/get-started:

Get started
===========

The upgrade process varies depending on where your database is hosted.

.. _db-upgrade/online:

Online (SaaS)
-------------

If you are hosted Online, please check our :ref:`instruction page for our Online (SaaS) customers
<upgrade_button>`.

.. _db-upgrade/odoo-sh:

Odoo.sh
-------

If you are Odoo.sh hosted, check our :doc:`specific instructions to be able to upgrade

> <../odoo_sh/advanced/upgrade_your_database>`.

Reviewer 2: It might be worth doing the same in those docs and refers to this one if customers are on premise.

Reviewer 3: good idea, but if we could already push this documentation in prod and improve it afterward, it would be great

Reviewer 2: That's really not a big piece of work. If you could tell @jcs-odoo in which doc pages (I assume the same that you referenced) those references should go, this would be a matter of minutes to have them in this PR too.

Author: @avg-odoo 
if you send me the docs and where to add the link by the end of the day, it can be added to this PR today indeed.
In the meanwhile, I already do and push the other changes :)

.. _db-upgrade/on-premise:

On-Premise
----------

There are two possibilities:

#. Via the interface of our `website form <https://upgrade.odoo.com>`_
#. | For technically-advanced users and partners, via the following command line (to be used on the
     machine where your database is hosted):
   | ``python <(curl -s beta.upgrade.odoo.com/upgrade) test -d <your db name> -t 14.0 -c <your

>      subscription code>``

Author: @avg-odoo  ?

   :align: center
   :alt: Selection of the "Testing" purpose in the upgrade form on Odoo

.. _db-upgrade/test-guidance:

Test guidance
~~~~~~~~~~~~~

Every business and organization has its own operational needs and will have to test its specific
Odoo instance respectively. However, we recommend you look at `the test scenario
<https://drive.google.com/open?id=1Lm4JqbsHBirB1wMi14UChoz_YHLjx5ec>`_ we created, a high-level idea
of what you should test and look out for.

.. todo:: change link "test scenario" once the related doc is published

.. _db-upgrade/test-assistance:

Assistance
~~~~~~~~~~

If you encounter issues or problems in the **test database**, please contact the Odoo Upgrade
Support:

#. Connect to our `Odoo Support page <https://www.odoo.com/help>`_.
#. Under the *Ticket Description* section, select *An issue related to my upgrade* ticket type.

   .. image:: media/db-upgrade-test-assistance.png
      :align: center
      :alt: Selection of "An issue related to my upgrade" as Ticket Type in the support form on Odoo

   .. warning::
      If you choose another *Ticket Description* type, the request will be redirected to another

>       team than the upgrade one and will slow down the processing and response time.

Author: I guess it isn't that obvious.

>       team than the upgrade one and will slow down the processing and response time.

Reviewer 2: No big deal either way.

   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
   our `Sales department <mailto:sales@odoo.com>`_.

.. _db-upgrade/steps-production:

The production launch
---------------------

.. _db-upgrade/production-live:

Production goes live
~~~~~~~~~~~~~~~~~~~~

The production upgrade request is when you decide to upgrade your current database with all your
production data (invoices, VAT returns, inventories, current orders) to a new version of your choice.

After your :ref:`tests <db-upgrade/steps-test>` are completed to your satisfaction, trigger the
request to upgrade your production database via our `website form <https://upgrade.odoo.com>`_.
Select *Production* purpose.

.. image:: media/db-upgrade-production-purpose.png
   :align: center
   :alt: Selection of the "Production" purpose in the upgrade form on Odoo

.. warning::
   | **Risks of going into production without first testing:**
   | No or insufficient testing may increase the chances of potential business interruptions (i.e.,
     no longer having the possibility to validate an action) or poor customer experiences (i.e., an

>      eCommerce website that does not work correctly) when launching the new version in production.

Author: it's a great suggestion, thanks

      :align: center
      :alt: "Detailed Description" field in the support form on Odoo

   .. warning::
      If you choose another *Ticket Description* type, the request will be redirected to another
      team than the upgrade one and will slow down the processing and response time.

.. _db-upgrade/faq:

FAQ
===

.. _db-upgrade/why:

Why upgrade?
------------

* You benefit from the latest features of the :ref:`new major version <db-upgrade/release-notes>`
  released by Odoo.
* If you are in an :ref:`unsupported version <db-upgrade/supported-versions>`, you get a new version
  with support.

.. _db-upgrade/when:

When to upgrade?
----------------

Whenever you want. You can make your upgrade request as soon as a new version is released on our
`website form <https://upgrade.odoo.com>`_.

.. _db-upgrade/availability:

Availability of the new version
-------------------------------

Please kindly note that as soon as we announce the release of a new major version (usually at the
end of year), the Upgrade Service team needs to adapt the upgrade scripts to it, which is why the
new version is not immediately available for existing databases.

.. _db-upgrade/finalization:

> Finalization of the upgrade (ETA)

Author: it works, but the brackets around ETA were not in the right place

```suggestion
Finalization of the upgrade (:abbr:`ETA (Estimated Time of Arrival)`)
```

and translate to the new structure. If you use multiple apps managing sensitive data (e.g.,
Accounting, Inventory, etc.), some cases may still require a human intervention, making the process
slower.

This is especially true during the first months following the release of a new major version, which
can significantly lengthen the upgrade delay.

In general, the ‘smaller’ the database, the quickest the upgrade. A single-user database that uses
only CRM will be processed faster than a multi-company, multi-user database that uses Accounting,
Sales, Purchase, and Manufacturing.

So, in a nutshell, what can impact your upgrade lead time?

* Source & targeted versions
* Installed apps
* Volume of data
* Amount of customization (models, fields, methods, workflows, reports, website, etc.)
* Installation of new apps or configuration changes after the start of the test phase

Usually, the delays experienced during the first request (waiting time between the time you
submitted your first request for a test upgrade) can generally give you an idea of the time to wait
for the production request.

.. _db-upgrade/custom-modules:

Upgrade of the custom modules
-----------------------------

> As stated in our :ref:`Odoo Enterprise Subscription Agreement, section "5.1 Standard charges"

Author: @avw-odoo 
I have a fix :)

```suggestion
As stated in our :doc:`../legal/terms/enterprise`, section :ref:`charges_standard`, this optional 
```

it still renders correctly according to the doc's title and section's title, like this
`As stated in our Odoo Enterprise Subscription Agreement, section 5.1 Standard charges, this optional service is subject to additional fees.`

> As stated in our :ref:`Odoo Enterprise Subscription Agreement, section "5.1 Standard charges"

Reviewer 2: odo will need to approve any change in legal/ but it's fine by me :+1:

> As stated in our :ref:`Odoo Enterprise Subscription Agreement, section "5.1 Standard charges"

Author: I didn't change anything in legal/ :)

<charges_standard>`, this optional service is subject to additional fees.

If you have a custom code, you can choose to have it upgraded by our services, by one of our
`partners <https://www.odoo.com/partners>`_ or you can do it yourself.

.. note::
   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
   our `Sales department <mailto:sales@odoo.com>`_.

.. _db-upgrade/editions-change:

Editions change (from Community to Enterprise)
----------------------------------------------

> An upgrade does not cover a change of `Editions <https://www.odoo.com/page/editions>`_

Reviewer 2: ```suggestion
An upgrade does not cover a change of `Editions <https://www.odoo.com/page/editions>`_.
```

.. note::
   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
   our `Sales department <mailto:sales@odoo.com>`_.

.. _db-upgrade/hosting-types-switch:

Switching the hosting types (Self-hosted vs Online vs Odoo.sh)
--------------------------------------------------------------

> An upgrade does not cover a change of `Hosting types <https://www.odoo.com/page/hosting-types>`_

Reviewer 2: ```suggestion
An upgrade does not cover a change of `Hosting types <https://www.odoo.com/page/hosting-types>`_.
```

Open the following link to get :doc:`more information about how to change your hosting type
<hosting_changes>`.

.. note::
   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact

>    our `Sales department <mailto:sales@odoo.com>`_.

Reviewer 2: Those notes are getting spammy. Is it really necessary to have so many of them?

Reviewer 3: yes they are absolutely necessary; This documentation is based on the recurring requests we receive in the upgrade support pipe, because there is a lot of confusion for Odoo customers and employees about what the upgrade covers and what the processes are if it is related to a change of hosting or editions;
So yes please, keep them

Reviewer 2: I understand but at the same time, that's a LOT of identical notes. It looks like advertising on some Parismatch blog post :stuck_out_tongue_closed_eyes:
In the end, I just make comments and suggestions here. Do what you want with them ;)

![unknown](https://user-images.githubusercontent.com/15896813/102357860-17706280-3faf-11eb-84c0-272e083f1acb.png)

Author: alright, we keep it, then :)
I replaced it in the code with substations

```
.. |assistance-contact| replace::
   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
   our `Sales department`_.
.. _Sales department: mailto:sales@odoo.com
```

.. _db-upgrade/release-notes:

Release Notes by version
------------------------

Open our `Release Note <https://www.odoo.com/page/release-notes>`_ page to get a summary of the new
features and improvements made in each version.

.. _db-upgrade/assistance:

Assistance
==========

.. _db-upgrade/contact:

Contact our Upgrade service support
-----------------------------------

Should you have any more questions about the upgrade, do not hesitate to send a message to `Odoo
Upgrade Team <mailto:upgrade@odoo.com>`_. We will be very pleased to answer it as soon as possible.

.. _db-upgrade/supported-versions:

Supported versions
------------------

Please note that Odoo provides support and bug fixing only for the three last major versions of Odoo.

> This is a factor to take into consideration before upgrading. If you are in an older version, we

Reviewer 2: ```suggestion
This is a factor to take into consideration before upgrading. If you are on an older version, we
```
Honestly don't know which one is correct. I'm highlighting it just in case.
